src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs

```diff
@@ -0,0 +1,1475 @@
+        }
+
+        public class ClassWithInvalidExtensionPropertyStringString
+        {
+            [JsonExtensionData]
+            public Dictionary<string, string> MyOverflow { get; set; }
+        }
+
+        public class ClassWithInvalidExtensionPropertyObjectString
+        {
+            [JsonExtensionData]
+            public Dictionary<DummyObj, string> MyOverflow { get; set; }
+        }
+
+        public class ClassWithInvalidExtensionPropertyStringJsonNode
+        {
+            [JsonExtensionData]
+            public Dictionary<string, JsonNode> MyOverflow { get; set; }
+        }
+
+        [Fact]
+#if BUILDING_SOURCE_GENERATOR_TESTS
+        [ActiveIssue("Compile-time errors thrown for these scenarios.")]
+#endif
+        public async Task ExtensionProperty_InvalidDictionary()
+        {
+            var obj1 = new ClassWithInvalidExtensionPropertyStringString();
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => await JsonSerializerWrapperForString.SerializeWrapper(obj1));
+
+            var obj2 = new ClassWithInvalidExtensionPropertyObjectString();
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => await JsonSerializerWrapperForString.SerializeWrapper(obj2));
+
+            var obj3 = new ClassWithInvalidExtensionPropertyStringJsonNode();
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => await JsonSerializerWrapperForString.SerializeWrapper(obj3));
+        }
+
+        public class ClassWithExtensionPropertyAlreadyInstantiated
+        {
+            public ClassWithExtensionPropertyAlreadyInstantiated()
+            {
+                MyOverflow = new Dictionary<string, object>();
+            }
+
+            [JsonExtensionData]
+            public Dictionary<string, object> MyOverflow { get; set; }
+        }
+
+        public class ClassWithExtensionPropertyAsObject
+        {
+            [JsonExtensionData]
+            public Dictionary<string, object> MyOverflow { get; set; }
+        }
+
+        public class ClassWithExtensionPropertyAsJsonElement
+        {
+            [JsonExtensionData]
+            public Dictionary<string, JsonElement> MyOverflow { get; set; }
+        }
+
+        public class ClassWithExtensionPropertyAsJsonObject
+        {
+            [JsonExtensionData]
+            public JsonObject MyOverflow { get; set; }
+        }
+
+        public class ClassWithExtensionPropertyAsSystemObject
+        {
+            [JsonExtensionData]
+            public object MyOverflow { get; set; }
+        }
+
+        public class ClassWithMultipleDictionaries
+        {
+            [JsonExtensionData]
+            public Dictionary<string, object> MyOverflow { get; set; }
+
+            public Dictionary<string, object> ActualDictionary { get; set; }
+        }
+
+        [Fact]
+#if BUILDING_SOURCE_GENERATOR_TESTS
+        [ActiveIssue("Compile-time errors thrown for these scenarios.")]
```

Comment:
Should we be logging issues for these disabled tests?

Reply:
Created https://github.com/dotnet/runtime/issues/58945. Will link it in source.